web_src/js/webcomponents/GiteaLocaleDate.js

```diff
@@ -0,0 +1,40 @@
+window.customElements.define('gitea-locale-date', class extends HTMLElement {
+  static observedAttributes = ['date', 'year', 'month', 'weekday', 'day'];
+
+  update = () => {
+    const year = this.getAttribute('year') ?? '';
+    const month = this.getAttribute('month') ?? '';
+    const weekday = this.getAttribute('weekday') ?? '';
+    const day = this.getAttribute('day') ?? '';
+    const lang = this.closest('[lang]')?.getAttribute('lang') ||
+      this.ownerDocument.documentElement.getAttribute('lang') ||
+      '';
+
+    // only extract the `yyyy-mm-dd` part
```

Comment:
```suggestion
    // only extract the `yyyy-mm-dd` part. When converting to Date, the date will be in UTC and when rendered
    // as locale date, will have a offset towards UTC added. We should eventually use `Temporal.PlainDate` here
    // to avoid needing to remove this offset: https://tc39.es/proposal-temporal/docs/plaindate.html
```